packages/core/data-transfer/types/utils.d.ts

```diff
@@ -2,63 +2,27 @@ import type { Readable, Writable, Duplex, Transform } from 'stream';
 import type { IDestinationProvider, ISourceProvider } from './providers';
 
 /**
- * Default signature for transfer rules' filter methods
+ * Define a transfer transform which will be used to intercept
+ * and potentially discard or modify the transferred data
  */
-type TransferRuleFilterSignature = (...params: unknown[]) => boolean;
-
-/**
- * Define a transfer rule which will be used to intercept
- * and potentially modify the transferred data
- */
-export interface ITransferRule<
-  T extends TransferRuleFilterSignature = TransferRuleFilterSignature
-> {
-  /**
-   * Filter method used to select which data should be transformed
-   */
-  filter?: T;
-  /**
-   * Transform middlewares which will be applied to the filtered data
-   */
-  transforms: StreamItem[];
-}
+export type TransferTransform<T> =
+  | {
+      filter: (data: T) => boolean;
+    }
+  | {
+      map: (data: T) => T;
+    };
 
 export type TransformFunction = (chunk: any, encoding?: string) => any;
 export type StreamItem = Stream | TransformFunction;
 type Stream = Readable | Writable | Duplex | Transform;
 
-export interface AddedDiff<T = unknown> {
-  kind: 'added';
-  path: string[];
-  type: string;
-  value: T;
-}
-
-export interface ModifiedDiff<T = unknown, P = unknown> {
-  kind: 'modified';
-  path: string[];
-  types: [string, string];
-  values: [T, P];
-}
-
-export interface DeletedDiff<T = unknown> {
-  kind: 'deleted';
-  path: string[];
-  type: string;
-  value: T;
-}
-
-export type Diff = AddedDiff | ModifiedDiff | DeletedDiff;
-
-export interface Context {
-  path: string[];
-}
 export type TransferStage = 'entities' | 'links' | 'assets' | 'schemas' | 'configuration';
 
 export interface ITransferResults<S extends ISourceProvider, D extends IDestinationProvider> {
   source?: S['results'];
   destination?: D['results'];
-  engine?: ITransferResults;
+  engine?: unknown;
```

Comment:
'unknown' is definitely better than whatever I was attempting to do before 😂 I think something got mixed up in the refactoring.

But this value is what is directly passed to the cli table builder, so we should try to define it. It's something like:
```
[key: TransferStage]: {
  bytes: number;
  count: number;
  aggregates: {
    [key: string]: {
      bytes: number;
      count: number;
    }
  }
}
```